src/modules/roc_packet/target_posix/roc_packet/address.cpp

```diff
@@ -129,7 +129,8 @@ bool Address::operator==(const Address& other) const {
         break;
 
     case AF_INET6:
-        if (sa_.addr6.sin6_addr.s6_addr != other.sa_.addr6.sin6_addr.s6_addr) {
+        if (memcmp(sa_.addr6.sin6_addr.s6_addr,
+                   other.sa_.addr6.sin6_addr.s6_addr, sizeof(in6_addr)) != 0) {
```

Comment:
There is a small problem here. `sa_.addr6.sin6_addr.s6_addr` is not of type in6_addr, `sa_.addr6.sin6_addr` is. We either need:

```
memcmp(sa_.addr6.sin6_addr.s6_addr, other.sa_.addr6.sin6_addr.s6_addr,
    sizeof(sa_.addr6.sin6_addr.s6_addr)
```

or:

```
memcmp(&sa_.addr6.sin6_addr, &other.sa_.addr6.sin6_addr,
    sizeof(in6_addr)
```

I think the first variant is preferred, because:

* it specified in POSIX that there a field s6_addr of type uint8_t[16], so it's ok to rely that it's present and that it's an array
* it's not specified in POSIX whether other auxiliary fields are present, and if they are, we likely don't want to include them into comparison

(see http://pubs.opengroup.org/onlinepubs/009695399/basedefs/netinet/in.h.html)

Reply:
Fixed